src/iop/demosaic.c

```diff
@@ -2480,6 +2485,47 @@ static void passthrough_monochrome(float *out, const float *const in, dt_iop_roi
   }
 }
 
+static void blk4_monochrome(float *out, const float *const in, dt_iop_roi_t *const roi_out,
+                                   const dt_iop_roi_t *const roi_in, const uint32_t filters, const uint32_t rgb_filter)
+{
+  assert(roi_in->width >= roi_out->width);
+  assert(roi_in->height >= roi_out->height);
+
+  const float rgb_pars[4][4] = {
+    { 0.2225f, 0.7169f / 2.0f, 0.0606f, 0.0f }, // luminance
+    { 0.5f, 0.5f / 2.0f, 0.0f, 0.0f },          // yellow
+    { 1.0f, 0.0f, 0.0f, 0.0f },                 // red
+    { 0.3333f, 0.3333f / 2.0f, 0.3333f, 0.0f } // equal
+  };
+
+#ifdef _OPENMP
+#pragma omp parallel for default(none) \
+  dt_omp_firstprivate(in, roi_out, roi_in) \
+  shared(out, rgb_pars, filters,rgb_filter) \
```

Comment:
`rgb_pars, filters, rgb_filter` should be `first_private`. This is false sharing if I'm not mistaken (aka performance leak). `out` could be `first_private` as well, but I'm less sure about that.

Reply:
By good chance you are right, although on my system there was no measurable performance gain.